packages/flutter/lib/src/material/toggle_button.dart

```diff
@@ -0,0 +1,1024 @@
+    this.borderRadius = const BorderRadius.all(Radius.circular(0.0)),
+    this.borderWidth = 1.0,
+    Widget child,
+  }) : super(key: key, child: child);
+
+  /// Creates a toggle buttons theme that controls the color and style
+  /// parameters for [ToggleButtons], and merges in the current toggle buttons
+  /// theme, if any.
+  ///
+  /// The [child] argument must not be null.
+  static Widget merge({
+    Key key,
+    Color color,
+    Color activeColor,
+    Color disabledColor,
+    Color fillColor,
+    Color focusColor,
+    Color highlightColor,
+    Color hoverColor,
+    Color splashColor,
+    Color borderColor,
+    Color activeBorderColor,
+    Color disabledBorderColor,
+    BorderRadius borderRadius = const BorderRadius.all(Radius.circular(0.0)),
+    double borderWidth = 1.0,
+    @required Widget child,
+  }) {
+    assert(child != null);
+    return Builder(
+      builder: (BuildContext context) {
+        final ToggleButtonsTheme parent = ToggleButtonsTheme.of(context);
+        return ToggleButtonsTheme(
+          key: key,
+          color: color ?? parent.color,
+          activeColor: activeColor ?? parent.activeColor,
+          disabledColor: disabledColor ?? parent.disabledColor,
+          fillColor: fillColor ?? parent.fillColor,
+          focusColor: focusColor ?? parent.focusColor,
+          highlightColor: highlightColor ?? parent.highlightColor,
+          hoverColor: hoverColor ?? parent.hoverColor,
+          splashColor: splashColor ?? parent.splashColor,
+          borderColor: borderColor ?? parent.borderColor,
+          activeBorderColor: activeBorderColor ?? parent.activeBorderColor,
+          disabledBorderColor: disabledBorderColor ?? parent.disabledBorderColor,
+          borderRadius: borderRadius ?? parent.borderRadius,
+          borderWidth: borderWidth ?? parent.borderWidth,
+          child: child,
+        );
+      },
+    );
+  }
+
+  /// The color for [Text] and [Icon] widgets.
+  ///
+  /// If [selected] is set to false and [onPressed] is not null, this color will be used.
+  final Color color;
+
+  /// The color for [Text] and [Icon] widgets.
+  ///
+  /// If [selected] is set to true and [onPressed] is not null, this color will be used.
+  final Color activeColor;
+
+  /// The color for [Text] and [Icon] widgets if the button is disabled.
+  ///
+  /// If [onPressed] is null, this color will be used.
+  final Color disabledColor;
+
+  /// The fill color for selected toggle buttons.
+  final Color fillColor;
+
+  /// The color to use for filling the button when the button has input focus.
+  final Color focusColor;
+
+  /// The highlight color for the button's [InkWell].
+  final Color highlightColor;
+
+  /// The splash color for the button's [InkWell].
+  final Color splashColor;
+
+  /// The color to use for filling the button when the button has a pointer hovering over it.
+  final Color hoverColor;
+
+  /// The border color to display when the toggle button is selected.
```

Comment:
I was struggling to determine the best property names to use, since the Material specs refer to "active" as "not disabled", as opposed to "active and selected". I think perhaps it could be better to adjust `color` and `borderColor` to be `selectedColor` and `selectedBorderColor`. That way, there is no confusion?

Reply:
Actually, would disabled/enabled/selected make more sense?

Reply:
Resolved offline to use the following convention:
foo -                  enabled state
selectedFoo -   selected state
disabledFoo -   disabled state